x-pack/plugin/security/src/test/java/org/elasticsearch/test/SecuritySettingsSource.java

```diff
@@ -130,6 +131,7 @@ public Settings nodeSettings(int nodeOrdinal) {
         writeFile(xpackConf, "users_roles", configUsersRoles());
 
         Settings.Builder builder = Settings.builder()
+                .put(Environment.PATH_HOME_SETTING.getKey(), home)
```

Comment:
In the newly added test, I create an Environment object (https://github.com/elastic/elasticsearch/pull/54999/files/60788f10dd3cb14e7d1a904174ed57c444ba2113#diff-985c60881de972e053b8aaff38e27168R48) which requires path.home to be set in the settings. The intention of the code in the security settings source was to prepare the config directory inside the home directory but the home directory was not specified in the settings from the configuration source (SecuritySettingsSource). The test framework would set the path.home setting through the use of default settings in InternalTestCluster before building the node so that's how it worked before this change.

Reply:
ah ok I see. `nodeSettings` is called before new `InternalTestCluster` is created and now `path.home` is needed inside `nodeSettings`. So it must be set eariler on. Thanks.